### PR TITLE
feat(SideNav): Add info in Changes menu item when user is disconnected

### PR DESCRIPTION
**Pull request by xrutayisire**
## Context

- Completes DT-1503

## The Solution

- Create a new component, "ChangesRightElement" to better manage that information and the conditions
- Added tests for that
- Increase a bit the max-width so it fit 
  - The max-width is strictly for the ellipsis as you can see, so no problem to increase it
  - max-width depend on the text that still need to be validated

## Impact / Dependencies

- No impact

## Checklist before requesting a review
- [x] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

## Preview

Screenshot 2023-07-20 at 17 25 50

### CODE REVIEW

File: packages/slice-machine/components/AppLayout/Navigation/ChangesRightElement.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { FC } from "react";
+
+import { useUnSyncChanges } from "@src/hooks/useUnSyncChanges";
+import { RightElement } from "@src/components/SideNav/SideNav";
+import { AuthStatus } from "@src/modules/userContext/types";
+import { useNetwork } from "@src/hooks/useNetwork";
+
+export const ChangesRightElement: FC = () => {
+  const isOnline = useNetwork();
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/AppLayout/Navigation/ChangesRightElement.tsx:9]: `useUnsyncedChanges` also returns `isOnline`

File: packages/slice-machine/components/AppLayout/Navigation/index.test.tsx [hunk 2]
```diff
@@ -133,6 +156,95 @@ describe("Side Navigation", () => {
...
+    expect(await screen.findByText("Changes")).toBeVisible();
+    const changesItem = screen
+      .getByText("Changes")
+      .closest("a") as HTMLAnchorElement;
+
+    expect(within(changesItem).getByText("Login required")).toBeVisible();
+  });
+
+  test("should display the information that user is disconnected on the 'Changes' item when offline", async () => {
+    vi.spyOn(navigator, "onLine", "get").mockReturnValueOnce(false);
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/AppLayout/Navigation/index.test.tsx:215]: hmm, that's an odd edge case, as they could have a valid login token but there's no why to check it

File: packages/slice-machine/src/components/SideNav/SideNav.css.ts [hunk 3]
```diff
@@ -316,7 +316,7 @@ export const rightElementText = style([
     alignSelf: "baseline",
     fontSize: "12px",
     lineHeight: "16px",
-    maxWidth: "68px",
+    maxWidth: "82px",
```
Comment (MarcMcIntosh) [inline packages/slice-machine/src/components/SideNav/SideNav.css.ts:319]: That's pretty big right?

Comment (xrutayisire) [reply packages/slice-machine/src/components/SideNav/SideNav.css.ts:319]: It's for the text "Login required" so the ellipsis don't trigger.
I tested it for the version ad it works well with our alpha one

File: packages/slice-machine/components/AppLayout/Navigation/index.test.tsx (hunk 2 above)
Comment (xrutayisire) [reply packages/slice-machine/components/AppLayout/Navigation/index.test.tsx:215]: True, but I did the same stuff as we are doing in the Changes page, so if it's offline you cannot see the page.

File: packages/slice-machine/components/AppLayout/Navigation/ChangesRightElement.tsx (hunk 1 above)
Comment (xrutayisire) [reply packages/slice-machine/components/AppLayout/Navigation/ChangesRightElement.tsx:9]: What is the best for you, I used it from there at the beginning but since it's coming from useNetwork, I kept it like this.
When we will rework useUnsyncedChanges it could simplify the hook. Each hook should have separate concern and isOnline don't look too much related. WDYT?

Comment (MarcMcIntosh) [reply packages/slice-machine/components/AppLayout/Navigation/ChangesRightElement.tsx:9]: Is there rework to be done on useUnsyncedChanges?.... that may change things

**Review by MarcMcIntosh — APPROVED** [review]

File: packages/slice-machine/components/AppLayout/Navigation/ChangesRightElement.tsx (hunk 1 above)
Comment (xrutayisire) [reply packages/slice-machine/components/AppLayout/Navigation/ChangesRightElement.tsx:9]: Yes for me, we should try to not mix everything, there is no direct link between useUnsyncedChanges and useNetwork. It create a bit of a all in one hook.

Comment (xrutayisire) [reply packages/slice-machine/components/AppLayout/Navigation/ChangesRightElement.tsx:9]: But we will see later on :)
